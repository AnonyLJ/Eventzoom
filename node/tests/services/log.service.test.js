import logModel from '../../src/models/log.model';
import logService from '../../src/services/log.service';

// Mock save. Mongoose save returns the Id of the document saved
const mockSave = jest.fn();
jest.mock('../../src/models/log.model', () => jest.fn().mockImplementation(() => ({ save: mockSave })));


describe('testing logOccurence', () => {
	it('should run successfully', async () => {
		mockSave.mockImplementation(async () => ({ _id: 5 }));
		await logService.logOccurence('test', 'test');
		expect(logModel.mock.calls[0]).toMatchSnapshot();
		expect(mockSave.mock.calls.length).toEqual(1);
	});
});
