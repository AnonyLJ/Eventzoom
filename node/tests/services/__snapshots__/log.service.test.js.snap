// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing logOccurence should run successfully 1`] = `
Array [
  Object {
    "data": "test",
    "event": "test",
    "referencedEvent": null,
  },
]
`;
